rest-api/dao/biobank_order_dao.py

```diff
@@ -265,8 +276,43 @@ def from_client_json(self, resource_json, participant_id=None, client_id=None):
           % (participant_id, resource.subject, self._participant_id_to_subject(participant_id)))
     self._add_identifiers_and_main_id(order, resource)
     self._add_samples(order, resource)
+    if resource.status and not resource.amendedInfo:
+      self._handle_new_status(order, resource)
+    if resource.amendedReason:
+      order.amendedReason = resource.amendedReason
+    if resource.amendedInfo:
+      order.amendedUsername, order.amendedSiteId = self._parse_handling_info(resource.amendedInfo)
+    order.version = expected_version
     return order
 
+  @classmethod
+  def _handle_new_status(cls, order, resource):
+    site_dao = SiteDao()
+    order.status = resource.status
+    if order.status.upper() == 'CANCELLED':
```

Comment:
Can we use enum value names for these instead?

Reply:
You can use BiobankOrderStatus.CANCELLED.name or str(BiobankOrderStatus.CANCELLED), and compare that to order.status. Or you can make a BiobankOrderStatus out of order.status here, and compare the enums.

(Comparisons of strings and enums directly will fail.)

Reply:
function removed as per above.